2010-05-26  Michael Tautschnig  <mt@debian.org>

	* subroutines: Fix eval_cmdline failures when finetuning NFS options (thanks
		Thorsten Glaser <tg@debian.org> and Waldemar Brodkorb
		<fai@waldemar-brodkorb.de>)
Index: trunk/lib/subroutines
===================================================================
--- trunk.orig/lib/subroutines
+++ trunk/lib/subroutines
@@ -704,12 +704,16 @@
 
     echo -n "Kernel currently running: "
     uname -rsmo
-    echo -n "Kernel parameters: "; cat /proc/cmdline
+    eval "cmdline=($(</proc/cmdline))"
+    echo "Kernel parameters: ${cmdline[*]}"
     shopt -s extglob
-    for word in $(cat /proc/cmdline) ; do
+    for word in "${cmdline[@]}"; do
 	case $word in
 	    *([a-zA-Z0-9_])=*)
-		eval "export $word"
+		varname=${word%%=*}
+		word=${word#*=}
+		eval $varname=\$word
+		eval export $varname
 		;;
 	esac
     done
